packages/augur-core/source/contracts/matic/AugurPredicate.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity 0.5.10;
+pragma experimental ABIEncoderV2;
+
+import { Registry } from "ROOT/matic/Registry.sol";
+import { BytesLib } from "ROOT/matic/libraries/BytesLib.sol";
+import { RLPReader } from "ROOT/matic/libraries/RLPReader.sol";
+
+import { IShareToken } from "ROOT/reporting/IShareToken.sol";
+import { ShareToken } from "ROOT/reporting/ShareToken.sol";
+import { IMarket } from "ROOT/reporting/IMarket.sol";
+import { IExchange } from "ROOT/external/IExchange.sol";
+import { ZeroXTrade } from "ROOT/trading/ZeroXTrade.sol";
+import { IAugurTrading } from "ROOT/trading/IAugurTrading.sol";
+import { IAugur } from "ROOT/IAugur.sol";
+
+contract AugurPredicate {
+    using RLPReader for bytes;
+    using RLPReader for RLPReader.RLPItem;
+
+    Registry public registry;
+
+    IAugur public augur;
+    ShareToken public shareToken;
+    ZeroXTrade public zeroXTrade;
+
+    struct ExitData {
+        address shareToken;
+        address cash;
+    }
+
+    mapping(uint256 => ExitData) public lookupExit;
+
+    function initialize(IAugur _augur, IAugurTrading _augurTrading) public /* @todo beforeInitialized */ {
+        // endInitialization();
+        augur = _augur;
+        // This ShareToken is the real slim shady
+        shareToken = ShareToken(_augur.lookup("ShareToken"));
+        zeroXTrade = ZeroXTrade(_augurTrading.lookup("ZeroXTrade"));
+    }
+
+    /**
+    * @dev Start exit with a zeroX trade
+    * @param _taker Order filler
+    */
+    function trade(
+        uint256 _requestedFillAmount,
+        address _affiliateAddress,
+        bytes32 _tradeGroupId,
+        IExchange.Order[] memory _orders,
+        bytes[] memory _signatures,
+        address _taker
+    )
+        public
+        payable
+    {
+        // @todo Handle case where the exitor is exiting with a trade filled by someone else (exitor had a signed order)
+        require(
+            _taker == msg.sender,
+            "Exitor is not the order taker"
+        );
+        ZeroXTrade.AugurOrderData memory _augurOrderData = zeroXTrade.parseOrderData(_orders[0]);
+        uint256 exitId = getExitId(_augurOrderData.marketAddress, msg.sender);
+        if (lookupExit[exitId].shareToken == address(0x0)) {
+            initializeForExit(exitId, _augurOrderData.marketAddress);
+        }
+        zeroXTrade.trade.value(msg.value)(
```

Comment:
Seems like no order validation is occurring here. All refs to the exchange are gone that is.